### PR TITLE
Initial work to sync playback to master

### PR DESCRIPTION
**Pull request by robhuzzey**
This PR should address issue: https://github.com/Festify/app/issues/124

I've not used typescript before so hopefully this is appropriate code... please test the hell out of this as I've been unable to test with multiple clients.

Hope this helps :)

### CODE REVIEW

File: src/reducers/view-settings.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export default function(
         tracksLoadInProgress: false,
         tracksToLoad: 0,
         tracksLoaded: 0,
+        syncedPlayback: false,
```
Comment (robhuzzey) [inline src/reducers/view-settings.ts]: Gah, just seen this isn't alphabetical... will fix.

File: src/views/party-settings.ts [hunk 2]
```diff
@@ -127,6 +130,14 @@ const PartySettings = (props: PartySettingsProps & PartySettingsDispatch) => htm
             Display "Ken Burns" background in TV mode
         </paper-checkbox>
 
+        <br />
+
+        <paper-checkbox checked="${props.syncedPlayback}"
+                        on-checked-changed="${ev => props.syncPlayback((ev.target as HTMLInputElement).checked)}"
+                        title="Syncs your playback with the party host.">
```
Comment (robhuzzey) [inline src/views/party-settings.ts]: I've only just realised that this setting is only seen by the host so the wording here seems odd... any suggestions? Maybe "sync your playback with the listeners"?

Comment (NeoLegends) [reply src/views/party-settings.ts]: Umm, this might need a fundamentally different approach, UX wise. Maybe we should enable the play button for parties where a host is playing, so that other people can start their instance of Festify in slave mode? Currently the button is disabled for everyone who doesn't have admin access to the party / is host.

This would require quite some refactoring in player handling, though.

File: src/sagas/local-player.ts [hunk 3]
```diff
@@ -108,23 +108,22 @@ function* handlePlaybackStateChange(
...
-        ]);
+    yield call(
+        playTrack,
+        currentTrack.reference.id,
+        deviceId,
+        syncPlayback ? currentParty.playback.last_position_ms : 0,
+    )
+
+    if (spotifyState && spotifyState.track_window.current_track.id === currentTrack.reference.id) {
+        yield call(markTrackAsPlayed, partyId, currentTrack.reference)
```
Comment (NeoLegends) [inline src/sagas/local-player.ts]: @robhuzzey Were you able to test the changes in here? The playback code has to handle lots of edge cases (thats why it seems so badly written / unnecessarily complicated) and I'm not sure whether this isn't breaking some things. :)

(did't get the chance to test this yet, will do next week)

File: src/views/party-settings.ts (hunk 2 above)
Comment (robhuzzey) [reply src/views/party-settings.ts]: Yeah, that was my concern as the UX doesn't feel right to be honest.

File: src/views/party-settings.ts [hunk 4]
```diff
@@ -127,6 +130,14 @@ const PartySettings = (props: PartySettingsProps & PartySettingsDispatch) => htm
             Display "Ken Burns" background in TV mode
         </paper-checkbox>
 
+        <br />
```
[the remark this answers is not in the document]
Comment (robhuzzey) [reply src/views/party-settings.ts]: gah, yep... this was me being lazy ;)

File: src/sagas/local-player.ts (hunk 3 above)
Comment (robhuzzey) [reply src/sagas/local-player.ts]: I wasn't able to test it properly as I don't even have 2 machines to try it out on. I've asked a few people at work to give it a test with me when they have time....

**Comment by robhuzzey** [comment]
@NeoLegends in chatting to a few people about this work & getting help testing I was directed to this app that pretty much does exactly what I was describing with synced playback: https://www.jqbx.fm/ Have you seen it yet?
